res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -0,0 +1,3157 @@
+        let motorWindDownTimer = 0;
+        const motorWindDownTimerCallback = () => {
+            engine.stopTimer(motorWindDownTimer);
+            motorWindDownTimer = 0;
+        };
+        this.turntableButton = UseMotors ? new Button({
+            deck: this,
+            input: function(press) {
+                if (press) {
+                    this.deck.reverseButton.loopModeOff(true);
+                    this.deck.fluxButton.loopModeOff(true);
+                    if (this.deck.wheelMode === wheelModes.motor) {
+                        this.deck.wheelMode = wheelModes.vinyl;
+                        motorWindDownTimer = engine.beginTimer(MotorWindDownMilliseconds, motorWindDownTimerCallback, true);
+                        engine.setValue(this.group, "scratch2_enable", false);
+                    } else {
+                        this.deck.wheelMode = wheelModes.motor;
+                        const group = this.group;
+                        engine.beginTimer(MotorWindUpMilliseconds, function() {
+                            engine.setValue(group, "scratch2_enable", true);
+                        }, true);
+                    }
+                    this.outTrigger();
+                }
+            },
+            outTrigger: function() {
+                const motorOn = this.deck.wheelMode === wheelModes.motor;
+                this.send(this.color + (motorOn ? this.brightnessOn : this.brightnessOff));
+                const vinylModeOn = this.deck.wheelMode === wheelModes.vinyl;
+                this.deck.jogButton.send(this.color + (vinylModeOn ? this.brightnessOn : this.brightnessOff));
+            },
+        }) : undefined;
+        this.jogButton = new Button({
+            deck: this,
+            input: function(press) {
+                if (press) {
+                    this.deck.reverseButton.loopModeOff(true);
+                    this.deck.fluxButton.loopModeOff(true);
+                    if (this.deck.wheelMode === wheelModes.vinyl) {
+                        this.deck.wheelMode = wheelModes.jog;
+                    } else {
+                        if (this.deck.wheelMode === wheelModes.motor) {
+                            motorWindDownTimer = engine.beginTimer(MotorWindDownMilliseconds, motorWindDownTimerCallback, true);
+                        }
+                        this.deck.wheelMode = wheelModes.vinyl;
+                    }
+                    engine.setValue(this.group, "scratch2_enable", false);
+                    this.outTrigger();
+                }
+            },
+            outTrigger: function() {
+                const vinylOn = this.deck.wheelMode === wheelModes.vinyl;
+                this.send(this.color + (vinylOn ? this.brightnessOn : this.brightnessOff));
+                const motorOn = this.deck.wheelMode === wheelModes.motor;
+                this.deck.turntableButton.send(this.color + (motorOn ? this.brightnessOn : this.brightnessOff));
```

Comment:
this will break if `turntableButton` is `undefined` (which is the case if `UseMotors` is `false`). Spotted while going through the failing tests.